authz/grpc_audit_end2end_test.go

```diff
@@ -0,0 +1,332 @@
+/*
```

Comment:
Nit: let's see if we can rename this file to something shorter.  At least remove the `grpc_` prefix since this is the grpc repo and it's unnecessary.  I'm not sure `end2end` is all that necessary, either, especially since these are the only audit logging tests.

Is this feature "auditing" or "audit logging"?  If the former: `audit_test.go` seems fine.  If the latter: `audit_logging_test.go` or `audit_log_test.go` or something like that works.